bitcoinspv/bootstrap_test.go

```diff
@@ -0,0 +1,89 @@
+package bitcoinspv
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/gonative-cc/relayer/bitcoinspv/clients"
+	"github.com/gonative-cc/relayer/bitcoinspv/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+func TestBootstrapRelayer(t *testing.T) {
+	ctx := context.Background()
+	const latestHeight = 90
+	const latestFinalized = latestHeight - confirmationDepth + 1
+
+	t.Run("successful bootstrap", func(t *testing.T) {
+		r, btcClient, lcClient := setupTest(t)
+
+		btcClient.On("GetBTCTipBlock").Return(&chainhash.Hash{}, int64(latestHeight), nil)
+		lcClient.On("GetLatestBlockInfo", ctx).Return(&clients.BlockInfo{
+			Height: latestHeight,
+		}, nil)
+		blocks := make([]*types.IndexedBlock, confirmationDepth)
+		for i := 0; i < confirmationDepth; i++ {
+			blocks[i] = &types.IndexedBlock{
+				BlockHeader: &wire.BlockHeader{},
+				BlockHeight: int64(i + latestFinalized),
+			}
+		}
+		btcClient.On("GetBTCTailBlocksByHeight", int64(latestFinalized), false).Return(blocks, nil)
+		lcClient.On("ContainsBlock", mock.Anything, mock.Anything).Return(true, nil)
+		btcClient.On("SubscribeNewBlocks").Return()
+
+		err := r.bootstrapRelayer(ctx, false)
+		assert.NoError(t, err)
+	})
+
+	t.Run("failed BTC sync", func(t *testing.T) {
+		r, btcClient, _ := setupTest(t)
+
+		expectedErr := errors.New("btc sync failed")
+		btcClient.On("GetBTCTipBlock").Return(nil, int64(0), expectedErr)
+
+		err := r.bootstrapRelayer(ctx, false)
+		assert.ErrorIs(t, err, expectedErr)
+	})
+
+	t.Run("failed light client sync", func(t *testing.T) {
+		r, btcClient, lcClient := setupTest(t)
+		btcClient.On("GetBTCTipBlock").Return(&chainhash.Hash{}, int64(latestFinalized), nil)
+
+		expectedErr := errors.New("light client sync failed")
+		lcClient.On("GetLatestBlockInfo", mock.Anything).Return(nil, expectedErr)
+
+		err := r.bootstrapRelayer(ctx, false)
+		assert.ErrorIs(t, err, expectedErr)
+	})
+}
+
+func TestWaitForBTCCatchup(t *testing.T) {
+	t.Run("successful catchup", func(t *testing.T) {
+		r, btcClient, lcClient := setupTest(t)
+		ctx := context.Background()
+
+		// Set shorter ticker for test
+		originalTicker := bootstrapSyncTicker
+		bootstrapSyncTicker = 100 * time.Millisecond
+		defer func() { bootstrapSyncTicker = originalTicker }()
```

Comment:
we can use t.Cleanup() for that as an alternative. I think its more suitable for tests

Reply:
I've sorted out in https://github.com/gonative-cc/relayer/pull/227 - it's in draft, it will be updated after this PR.